### PR TITLE
[Keystone] add rate-limit middleware

### PR DESCRIPTION
**Pull request by rajivmucheli**
Hi @Carthaca,

Please validate this PR.

Regards,
Rajiv

### CODE REVIEW

File: keystone/server/flask/core.py [hunk 1]
```diff
@@ -79,9 +79,20 @@
                       'config_file': '/etc/keystone/watcher.yaml',
                       'include_initiator_user_id_in_metric': 'true',
                       'include_target_project_id_in_metric': 'false'}),
+    # CCloud: add rate_limit middleware
+    _Middleware(namespace='rate_limit.middleware',
+                ep='rate_limit',
```
Comment (Carthaca) [inline keystone/server/flask/core.py:84]: the endpoint must be `rate-limit` see https://github.com/sapcc/openstack-rate-limit-middleware/blob/master/setup.cfg#L33-L39

File: keystone/server/flask/core.py [hunk 2]
```diff
@@ -79,9 +79,20 @@
...
+                conf={'config_file': '/etc/keystone/ratelimit.yaml',
+                      'service_type': 'identity',
+                      'rate_limit_by': 'initiator_project_id',
```
Comment (Carthaca) [inline keystone/server/flask/core.py:87]: this is the default - but please also test the results with `initiator_host_address`

**Review by Carthaca — CHANGES_REQUESTED** [review]
Hi, please have a look inline. Also you don't need to specify the configs if you use the default value (max_sleep_time_seconds,log_sleep_time_seconds,backend_port etc.)

**Comment by rajivmucheli** [comment]
https://github.com/sapcc/keystone/pull/9

**Pull request closed without merging**